pkg/vendir/fetch/git/sync.go

```diff
@@ -36,7 +41,7 @@ func (d Sync) Desc() string {
 	return fmt.Sprintf("%s@%s", d.opts.URL, ref)
 }
 
-func (d Sync) Sync(dstPath string, tempArea ctlfetch.TempArea) (ctlconf.LockDirectoryContentsGit, error) {
+func (d Sync) Sync(dstPath string, tempArea ctlfetch.TempArea, cache ctlcache.Cache) (ctlconf.LockDirectoryContentsGit, error) {
```

Comment:
Instead of passing `cache` as an argument to this function, I'd use the approach of the Imgpkg syncer: https://github.com/carvel-dev/vendir/blob/f59017e59730f530e866ae749a4da107db90eb81/pkg/vendir/fetch/image/sync.go#L15-L27

It carries the cache object within the `Sync` struct, making it available to all its functions. Then, there might a function `getBundleName` for getting the bundle name (if exists) from the cache. `getBundleName` function can be used in the `fetch` function directly.

These two measures will allow to keep signatures of the functions (except the `git.NewSync`) unchanged.

Reply:
I move the cache to NewSync.

Reply:
about getBundleName, if we want to avoid an extra argument to git.Retrieve, we would need to put it on the Git struct. And I have a feeling it is not the job of the Git type.